app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -49,17 +49,18 @@ def default_authn_context
 
   def link_identity_from_session_data
     IdentityLinker.new(
-      current_user,
-      current_issuer
-    ).link_identity(
-      ial: loa3_requested? ? 3 : 1
-    )
+      current_user, current_issuer
+    ).link_identity(ial: ial_level)
   end
 
   def identity_needs_verification?
     loa3_requested? && current_user.decorate.identity_not_verified?
   end
 
+  def ial_level
```

Comment:
So I think I get what this was either 3 or 2. LOA3 means that you have IAL2. It is confusing, but where we are describing "IAL level" we should be using 2 and where we are describing LOA we should be using 3.